packages/core/src/api/exporters/html/util/serializeBlocksExternalHTML.ts

```diff
@@ -53,16 +54,63 @@ export function serializeInlineContentExternalHTML<
     throw new UnreachableCaseError(blockContent.type);
   }
 
-  // We call the prosemirror serializer here because it handles Marks and Inline Content nodes nicely.
-  // If we'd want to support custom serialization or externalHTML for Inline Content, we'd have to implement
-  // a custom serializer here.
-  const dom = serializer.serializeFragment(Fragment.from(nodes), options);
+  // Check if any of the nodes are custom inline content with toExternalHTML
+  const doc = options?.document ?? document;
```

Comment:
nice to see this implemented, do we have unit tests for this?

also, would this close;
https://github.com/TypeCellOS/BlockNote/issues/1625
and https://github.com/TypeCellOS/BlockNote/issues/1658
?

Reply:
Does have unit tests (new test cases & snapshots added).

Yea, it'd close both. Will add to the description.